--- a/gas/config/tc-z80.c.orig	2012-11-13 10:36:40.000000000 +0000
+++ a/gas/config/tc-z80.c	2012-11-13 10:40:24.000000000 +0000
@@ -467,7 +467,7 @@
   if (ins_type & ins_err)
     error (_(p));
   else
-    as_warn (_(p));
+    as_warn ("%s", _(p));
 }
 
 static void
@@ -544,12 +544,11 @@
 {
   const char *p;
   int indir;
-  segT dummy;
 
   p = skip_space (s);
   op->X_md = indir = is_indir (p);
   input_line_pointer = (char*) s ;
-  dummy = expression (op);
+  expression (op);
   switch (op->X_op)
     {
     case O_absent:
@@ -705,7 +704,6 @@
 {
   char *p;
   int lo, hi;
-  fixS * fixp;
 
   p = frag_more (1);
   *p = val->X_add_number;
@@ -732,7 +730,7 @@
     }
   else
     {
-      fixp = fix_new_exp (frag_now, p - frag_now->fr_literal, 1, val,
+      fix_new_exp (frag_now, p - frag_now->fr_literal, 1, val,
 			  (r_type == BFD_RELOC_8_PCREL) ? TRUE : FALSE, r_type);
       /* FIXME : Process constant offsets immediately.  */
     }
